tests/std/tests/GH_000690_overaligned_function/test.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+
+#ifndef __cpp_aligned_new
+#error This test is only valid when C++17 over-aligned allocation is supported
+#endif
+
+#pragma warning(disable : 4324) // structure was padded due to alignment specifier
+
+struct alignas(2 * alignof(std::max_align_t)) overaligned_t {
+    char non_empty;
+
+    void operator()(const void* const storage, const std::size_t storage_size) const {
+        const std::uintptr_t storage_ptr_value = reinterpret_cast<std::uintptr_t>(storage);
+        const std::uintptr_t this_ptr_value    = reinterpret_cast<std::uintptr_t>(this);
+
+        // platform-specific behavior not covered by Standard C++, but fine for such test
+        assert(this_ptr_value < storage_ptr_value || this_ptr_value >= storage_ptr_value + storage_size);
+        assert(this_ptr_value % alignof(overaligned_t) == 0);
+    }
+};
+
+static_assert(sizeof(overaligned_t) < sizeof(std::function<void()>), "overaligned_t is not aligned as expected");

```

Comment:
I don't see the relationship between the condition and the message here. I suspect you really want to assert that `overaligned_t` would otherwise be small if it weren't overaligned, which isn't possible in a blackbox test. There's implementation-specific overhead (two pointers for us right now) that prevents `std::function` from directly storing an object the same size as `std::function`.

Reply:
Yes, correct. I've replaced it with a runtime check.

Reply:
Or do you think such check should be removed completely, and instead test should just try to put `2*alignof(max_align_t)` object into a function, and it does not matter if SFO for non-aligned version of object with the same size would have happened or not?

Reply:
I like the test that an otherwise-similar object with less strict alignment triggers SFO, it ensures that alignment - the variable we mean to test - is in fact the only variable.